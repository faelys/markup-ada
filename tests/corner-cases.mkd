# Newline autodetection failure

At this point the renderer hasn't encountered any newline, because of internals
of ATX header parsing. So in autodetection mode, no newline is actually output
between closing `h1` and opening `p`.

# Indentation

No indentation yields a normal paragraph.

 One space indentation also yields a normal paragraph.

  Two-space indentation is still a normal paragraph.

   Three-space indentation is on the edge but still a normal paragraph.

    Four-space indentation switches to code block.

	One-tab indentation is also a code block.

# Fuzzy ATX headers

These are not ATX headers but match some of the parsing steps:

###

###   

###  ###

########

# Definition List with a Single Block Description

term

:   Single-block description

break

term1
:   Standard description

term2

:   Single-block second description

# Discount image without title but with size indication

![alt text](http://example.com/img.png =150x100)

# Fuzzy HTML comments

None of those are valid HTML tags or comments:

<

<!

<!-

Following is an invalid HTML comment that is still accepted as such by
the parser.

<!-- -- -->

# Footer

Footer to prevent wierd stuff with temporarily-last lines.
