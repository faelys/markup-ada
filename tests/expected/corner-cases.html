<h1>Newline autodetection failure</h1><p>At this point the renderer hasn't encountered any newline, because of internals
of ATX header parsing. So in autodetection mode, no newline is actually output
between closing <code>h1</code> and opening <code>p</code>.</p>
<h1>Indentation</h1>
<p>No indentation yields a normal paragraph.</p>
<p> One space indentation also yields a normal paragraph.</p>
<p>  Two-space indentation is still a normal paragraph.</p>
<p>   Three-space indentation is on the edge but still a normal paragraph.</p>
<pre><code>Four-space indentation switches to code block.

One-tab indentation is also a code block.</code></pre>
<h1>Fuzzy ATX headers</h1>
<p>These are not ATX headers but match some of the parsing steps:</p>
<p>###</p>
<p>###   </p>
<p>###  ###</p>
<p>########</p>
<h1>Fuzzy HTML comments</h1>
<p>None of those are valid HTML tags or comments:</p>
<p>&lt;</p>
<p>&lt;!</p>
<p>&lt;!-</p>
<p>Following is an invalid HTML comment that is still accepted as such by
the parser.</p>
<!-- -- -->
<h1>Footer</h1>
<p>Footer to prevent wierd stuff with temporarily-last lines.</p>
